sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -428,6 +428,33 @@ object DataType {
     }
   }
 
+  /**
+   * Returns true if the two data types share the same shape by name, i.e. the names
+   * are the same, but the types don't need to be the same.
+   */
+  def equalsStructurallyByName(
```

Comment:
Open to another name. I was original thinking `equalsStructurally`, but that already exists and does basically the exact opposite of what I was going for hah